pallets/base-nft/src/lib.rs

```diff
@@ -231,39 +231,46 @@ impl<T: Config> Pallet<T> {
 
 	/// Transfer NFT(non fungible token) from `from` account to `to` account
 	pub fn transfer(
-		from: &T::AccountId,
-		to: &T::AccountId,
+		from: Vec<&T::AccountId>,
+		to: Vec<&T::AccountId>,
 		token: (T::ClassId, T::TokenId),
+		fraction_amount: u8,
 	) -> DispatchResult {
 		Tokens::<T>::try_mutate(token.0, token.1, |token_info| -> DispatchResult {
 			let mut info = token_info.as_mut().ok_or(Error::<T>::TokenNotFound)?;
-			ensure!(info.owner == *from, Error::<T>::NoPermission);
+			ensure!(
+				info.owners.iter().any(|v| from.iter().any(|w| w == &v)),
```

Comment:
I am wondering if there is a good way to reduce the complexity on this, instead of doing it with a nested loop. I had thought about a solution using a `HashMap` in two separate loops, but it seems like [Substrate explicitly does not allow that](https://substrate.dev/docs/en/knowledgebase/advanced/no-hash-collections).